### PR TITLE
Mqf

### PR DESCRIPTION
**Pull request by shokrof**
- [Yes ] Is it mergeable?
- [ ] `make test` Did it pass the tests?
I introduced some test cases in tests_qfstorage.py to test MQF. Some unrelated tests fail on my machine.
- [ ] `make clean diff-cover` If it introduces new functionality in
  `scripts/` is it tested?
- [ ] `make format diff_pylint_report cppcheck doc pydocstyle` Is it well
  formatted?
- [ No] Did it change the command-line interface? Only backwards-compatible
  additions are allowed without a major version increment. Changing file
  formats also requires a major version number increment.
- [ ] For substantial changes or changes to the command-line interface, is it
  documented in `CHANGELOG.md`? See [keepachangelog](http://keepachangelog.com/)
  for more details.
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] Do the changes respect streaming IO? (Are they
  tested for streaming IO?)

I just replaced cqf library with the mqf. New functions like resize, merge, subtract, and tags are oimplemented and tested in mqf library, but Cython wrapper needs tp be developed to expose these functions to khmer. 
You can find draft documentation for mqf at https://github.com/shokrof/MQF/tree/mqfDevelopmenet.
Also, Test cases can be good examples. check https://github.com/shokrof/MQF/tree/mqfDevelopmenet/tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -132,6 +132,7 @@ clean: FORCE
 	cd tests && rm -rf khmertest_* || true
 	rm -f pytests.xml
 	cd third-party/cqf && make clean || true
```
Comment (drtamermansour) [inline Makefile]: Should not we remove this line?

File: include/oxli/storage.hh [hunk 2]
```diff
@@ -421,7 +421,11 @@ public:
     // in the key (current value of size+8 is copied from the CQF example)
     // Final argument is the number of bits allocated for the value, which
     // we do not use.
-    qf_init(&cf, (1ULL << size), size+8, 0);
+    _supports_bigcount = true;
+    qf_init(&cf, (1ULL << size), size+8, 0,2,true,"",2038074761);
```
Comment (drtamermansour) [inline include/oxli/storage.hh]: There is no documentation for the initialization functions at mqfDevelopmenet!

Comment (drtamermansour) [reply include/oxli/storage.hh]: Also if I understand this correctly, "cf" is your variable name. Let's use "mf" instead to make it more representative.

File: scripts/abundance-dist.py [hunk 3]
```diff
@@ -88,6 +88,9 @@ def get_parser():
                         'do not exist or are empty.')
     parser.add_argument('-q', '--quiet', dest='quiet', default=False,
                         action='store_true')
+    parser.add_argument('-c', '--use-CQF', dest='useCQF', default=False,
```
Comment (drtamermansour) [inline scripts/abundance-dist.py]: I think we can use '--use-MQF' & 'useMQF' instead

File: scripts/load-into-counting.py [hunk 4]
```diff
@@ -101,6 +101,8 @@ def get_parser():
                         help='Overwrite output file if it exists')
     parser.add_argument('-q', '--quiet', dest='quiet', default=False,
                         action='store_true')
+    parser.add_argument('-c', '--use-CQF', dest='useCQF', default=False,
```
Comment (drtamermansour) [inline scripts/load-into-counting.py]: same comment as before

File: setup.cfg [hunk 5]
```diff
@@ -6,7 +6,7 @@ undef = NO_UNIQUE_RC
 # docker/Dockerfile
 # libraries = z,bz2
 ## if using system libraries
-include-dirs = include:third-party/zlib:third-party/bzip2:third-party/seqan/core/include:third-party/smhasher:third-party/cqf:third-party/rollinghash
+include-dirs = include:third-party/zlib:third-party/bzip2:third-party/seqan/core/include:third-party/smhasher:third-party/cqf:third-party/rollinghash:third-party/mqf
```
Comment (drtamermansour) [inline setup.cfg]: should not we remove third-party/cqf?

File: runExperimentAbundanceCQF.sh [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+subsetFile=$readsFile
+
+head -n $subset < $readsFile > $subsetFile 
+
+echo "Counting Normal"
+
+python3 scripts/load-into-counting.py   -x $sketchSize $sketchCQFFile $subsetFile
+
+
+echo "Counting CQF"
```
Comment (drtamermansour) [inline runExperimentAbundanceCQF.sh]: The file name, the echo messages, and the test folder are still referring to CQF instead of MQF.

**Comment by ctb** [comment]
thanks for getting this started, Mostafa!

**Pull request closed without merging**